src/Microsoft.AspNetCore.Mvc.Core/ForbidObjectResult.cs

```diff
@@ -0,0 +1,126 @@
+        /// specified authentication schemes.
+        /// </summary>
+        /// <param name="authenticationSchemes">The authentication schemes to challenge.</param>
+        /// <param name="value">Response payload.</param>
+        public ForbidObjectResult(object value, IList<string> authenticationSchemes)
+            : this(value, authenticationSchemes, properties: null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbidObjectResult"/> with the
+        /// specified <paramref name="properties"/>.
+        /// </summary>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <param name="value">Response payload.</param>
+        public ForbidObjectResult(object value, AuthenticationProperties properties)
+            : this(value, Array.Empty<string>(), properties)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbidObjectResult"/> with the
+        /// specified authentication scheme and <paramref name="properties"/>.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication schemes to challenge.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <param name="value">Response payload.</param>
+        public ForbidObjectResult(object value, string authenticationScheme, AuthenticationProperties properties)
+            : this(value, new[] { authenticationScheme }, properties)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbidObjectResult"/> with the
+        /// specified authentication schemes and <paramref name="properties"/>.
+        /// </summary>
+        /// <param name="authenticationSchemes">The authentication scheme to challenge.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <param name="value">Response payload.</param>
+        public ForbidObjectResult(object value, IList<string> authenticationSchemes, AuthenticationProperties properties)
+            : base(value)
+        {
+            AuthenticationSchemes = authenticationSchemes;
+            Properties = properties;
+        }
+
+        /// <summary>
+        /// Gets or sets the authentication schemes that are challenged.
+        /// </summary>
+        public IList<string> AuthenticationSchemes { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="AuthenticationProperties"/> used to perform the authentication challenge.
+        /// </summary>
+        public AuthenticationProperties Properties { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ExecuteResultAsync(ActionContext context)
```

Comment:
Overriding `ExecuteResultAsync` wouldn't format the object. You would have to call the base ExecuteResultAsync. @Tratcher since the documentation doesn't cover this - is it acceptable to continue writing to the response after calling `HttpContext.ForbidAsync`?

Reply:
Yes as long as you're careful to ensure that your auth provider does not write to the body. None of our providers would write to the body for Forbid, but Cookies would redirect so your content would not be shown.

Reply:
Hmm, OAuth and OIDC would forward Forbid to Cookies, so the only provider this would work for would be Bearer which is used for WebAPIs.

Reply:
I guess we could document it as such. It's kinda unfortunate this isn't as straightforward as some of the other status code returning object results.